sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs

```diff
@@ -103,22 +103,35 @@ protected override void Initialize(SonarAnalysisContext context)
                 c =>
                 {
                     var addExpression = (TBinaryExpressionSyntax)c.Node;
-                    if (!IsInCheckedContext(addExpression, c.SemanticModel))
-                    {
-                        return;
-                    }
                     var leftNode = GetLeftNode(addExpression);
+
+                    var isInCheckedContext = false;
+                    // SemanticModel calls are expensive
+                    var ranSemanticCheck = false;
```

Comment:
We don't need 2 variables for this. This can be nicely done with `Lazy<bool>` and the code will simplify a lot.
```
var isInCheckedContext = new Lazy<bool>(() => IsInCheckedContext(addExpression, c.SemanticModel));
```